src/datasets/filesystems/s3filesystem.py

```diff
@@ -5,10 +5,40 @@
 
 class S3FileSystem(s3fs.S3FileSystem):
     """
-    ``datasets.filesystems.S3FileSystem`` is a subclass of s3fs.S3FileSystem](https://s3fs.readthedocs.io/en/latest/api.html), which is a known
-    implementation of ``fsspec``. Filesystem Spec [FSSPEC](https://filesystem-spec.readthedocs.io/en/latest/?badge=latest) is a project to
-    unify various projects and classes to work with remote filesystems
-    and file-system-like abstractions using a standard pythonic interface.
+    `datasets.filesystems.S3FileSystem` is a subclass of [`s3fs.S3FileSystem`](https://s3fs.readthedocs.io/en/latest/api.html).
+
+    Users can use this class to access S3 as if it were a file system. It exposes a filesystem-like API (ls, cp, open, etc.) on top of S3 storage. Provide credentials either explicitly (`key=`, `secret=`) or with boto's credential methods. See botocore documentation for more information. If no credentials are availabe, use `anon=True`.
+
+    Args:
+        anon (`bool`, default to `False`): Whether to use anonymous connection (public buckets only). If False, uses the key/secret given,
+            or boto's credential resolver (client_kwargs, environment, variables, config files, EC2 IAM server, in that order).
+        key (`str`): If not anonymous, use this access key ID, if specified.
+        secret (`str`): If not anonymous, use this secret access key, if specified.
+        token (`str`): If not anonymous, use this security token, if specified.
+        use_ssl (`bool`, default to `True`): Whether to use SSL in connections to S3; may be faster without, but insecure. If `use_ssl` is
+            also set in `client_kwargs`, the value set in `client_kwargs` will take priority.
+        s3_additional_kwargs (`Dict`): Parameters that are used when calling s3 api methods. Typically used for things
+            like ServerSideEncryption.
+        client_kwargs (`Dict`): Parameters for the botocore client.
+        requester_pays (`bool`, default to `False`): If RequesterPays buckets are supported.
+        default_block_size (`int`): If given, the default block size value used for `open()`, if no specific value is given at all time.
+            The built-in default is 5MB.
+        default_fill_cache (`bool`, default to `True`): Whether to use cache filling with open by default. Refer to `S3File.open`.
+        default_cache_type (`str`, default to `bytes`): If given, the default cache_type value used for `open()`. Set to `none` if no
+            caching is desired. See fsspec's documentation for other available cache_type values. Default cache_type is `bytes`.
```

Comment:
No need to repeat it (already written in line above)?
```suggestion
            caching is desired. See fsspec's documentation for other available cache_type values.
```